core/services/relay/evm/write_target.go

```diff
@@ -69,7 +83,191 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
 		return nil, err
 	}
 
-	return targets.NewWriteTarget(logger.Named(lggr, "WriteTarget"), id, cr, cw, config.ForwarderAddress().String(), gasLimitDefault), nil
+	pollPeriod, err := commonconfig.NewDuration(2 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create poll period: %w", err)
+	}
+
+	// TODO: Unsure what the timeout should be, I don't see an EVM config that corresponds to this.
+	timeout, err := commonconfig.NewDuration(30 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create timeout: %w", err)
+	}
+
+	chainInfo, err := getChainInfo(chain.ID().Uint64())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain info: %w", err)
+	}
+
+	beholder, err := writetarget.NewMonitor(ctx, lggr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aptos WT monitor client: %+w", err)
+	}
+
+	opts := writetarget.WriteTargetOpts{
+		ID:     id,
+		Logger: lggr,
+		Config: writetarget.Config{
+			ConfirmerPollPeriod: pollPeriod,
+			ConfirmerTimeout:    timeout,
+		},
+		ChainInfo:        chainInfo,
+		Beholder:         beholder,
+		ChainService:     chain,
+		ContractReader:   cr,
+		ChainWriter:      cw,
+		ConfigValidateFn: evaluate,
+		NodeAddress:      "nil",
```

Comment:
?

Reply:
Fixed this https://github.com/smartcontractkit/chainlink/blob/036dff1d212c45d401cb96e6807baba54bf7417c/core/services/relay/evm/write_target.go#L120